queries/queries.go

```diff
@@ -79,6 +79,16 @@ type DistributedQuery struct {
 	Expiration    time.Time
 }
 
+// NodeQuery links a node to a query
+type NodeQuery struct {
+	ID        uint      `gorm:"primaryKey;autoIncrement"`
+	NodeID    uint      `gorm:"not null;index"`
+	QueryID   uint      `gorm:"not null;index"`
+	Status    string    `gorm:"type:varchar(10);default:'pending'"`
```

Comment:
We can also use ENUM type here for more consistency

`CREATE TYPE task_status AS ENUM ('pending', 'completed', 'error');`

Reply:
The ORM I am using (https://gorm.io/) supports ENUMs, so we can explore using that: https://techwasti.com/how-to-use-enums-with-gorm

Reply:
Just realize that SQLite does not support ENUM and we can use the char for now and update later if we see another bottleneck issue.